src/functions.cpp

```diff
@@ -141,10 +141,28 @@ void removeFile(std::string_view filename)
     }
 }
 
+/**
+ * \brief Get the current timestamp as a string.
+ *
+ * This function generates the current timestamp in the format
+ * "YYYYMMDD_HHMMSS".
+ *
+ * \return A string with the current timestamp.
+ */
+std::string getCurrentTimestamp()
+{
+    auto now = std::chrono::system_clock::now();
+    auto time_t_now = std::chrono::system_clock::to_time_t(now);
+    std::stringstream ss;
```

Comment:
Is it necessary to use `std::stringstream`? Just asking because `std::stringstream` sometimes is a pain to deal with and the syntax is horrible: `<<`, `>>`, etc. Is there a way to use `fmt::format` instead?